src/cargo/ops/cargo_generate_lockfile.rs

```diff
@@ -405,4 +399,25 @@ impl PackageDiff {
 
         changes.into_iter().map(|(_, v)| v).collect()
     }
+
+    /// For querying [`PackageRegistry`] for alternative versions to report to the user
+    pub fn alternatives_query(&self) -> Option<crate::core::dependency::Dependency> {
+        let package_id = [
+            self.added.iter(),
+            self.unchanged.iter(),
+            self.removed.iter(),
```

Comment:
`removed` is also included here. I don't get the reason reporting latest availables for removed crates.

Reply:
This function is more generic than that and exists just to find alternative versions for the current `PackageDiff`.  If a new design has a reason to look up alternatives for remove-only then I'd rather not surprise them by this not working,